📖  Add `ClusterClass` documentation for `ClusterTopology` controller (#5602)
kubernetes-sigs/cluster-api: docs/book/src/developer/architecture/controllers/cluster-topology.md

# `ClusterTopology` Controller

> A `ClusterTopology` controller reconciles the managed topology of a Cluster. 

Reviewer: ```suggestion
The `ClusterTopology` controller reconciles the managed topology of a Cluster. 
```
nit: there is just one

![Cluster Topology Controller Activity Diagram](../../../images/cluster-topology-controller.png)

Its main responsibilities are:

> 1. Extend the Cluster object to use ClusterClass for creating managed topologies. 

Reviewer 2: ```suggestion
1. Reconcile Clusters based on templates defined in a ClusterClass and managed topology. 
```

> 2. Create/Update/Delete managed topologies (this includes continuous reconciliation of the topology managed resources)o.

Reviewer: ```suggestion
2. Create/Update/Delete managed topologies (this includes continuous reconciliation of the topology managed resources).
```
nit

> 3. Introduce mechanisms to allow Cluster-specific customizations of a ClusterClass

Reviewer 2: ```suggestion
3. Reconcile Cluster-specific customizations of a ClusterClass.
```

The high level workflow of ClusterTopology controller is shown below.

![ClusterTopology Reconciller Component Diagram](../../../images/cluster-topology-reconciller.png)

> ### Implementations

Reviewer: ```suggestion
### Implementation
```
nit

Reviewer: Or maybe: "Additional information" ?

Reviewer 2: I think "additional information" is better, and we should try to move past the link to the proposal and put the implementation details here once ClusterClass is out of alpha.
